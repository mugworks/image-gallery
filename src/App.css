.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

img {
  height: 400px;
}

.thumbnail {
  margin-top: 50px;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  width: 60%;
  margin: 10px; 
}

.thumbnail-img {
  height: 100px;
  margin: 0px 25px;
}

.button {
  background-color: rgb(102, 76, 175); 
  border: none;
  color: white;
  padding: 15px 30px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin: 4px 12px;
  cursor: pointer;
}

.remove, .add {
  font-size: 12px;
  margin: 4px 2px;
  padding: 10px 20px;
}

.add {
  margin-left: 10px;
}

#table {
  font-family: "Trebuchet MS", Arial, Helvetica, sans-serif;
  border-collapse: collapse;
  width: 100%;
  margin-top: 40px;
}

#table td, #table th {
  border: 1px solid #ddd;
  padding: 8px;
}

#table tr:nth-child(even){background-color: #f2f2f2;}

#table tr:hover {background-color: #ddd;}

#table th {
  padding-top: 12px;
  padding-bottom: 12px;
  text-align: center;
  background-color: rgb(130, 117, 173);
  color: white;
}

#form {
  text-align: left;
  width: 60%;
  margin-top: 50px;
  margin-left: auto;
  margin-right: auto;
  border-radius: 5px;
  background-color: rgb(212, 202, 202);
  padding: 20px;
}

#form-input {
  width: 30%;
  background-color: #f2f2f2;
  color: white;
  padding: 14px;
  margin: 8px 0;
  border: none;
  border-radius: 4px;
  cursor: pointer;
}

#gallery-container {
  margin-left: auto;
  margin-right: auto;
  margin-top: 50px;
}

.arrow {
  border: solid black;
  border-width: 0 3px 3px 0;
  display: inline-block;
  padding: 3px;
}

.right {
  transform: rotate(-45deg);
  -webkit-transform: rotate(-45deg);
}

.left {
  transform: rotate(135deg);
  -webkit-transform: rotate(135deg);
}

.next, .previous{
  margin: 5px;
  padding: 5px;
  font-size: 14px;
}


