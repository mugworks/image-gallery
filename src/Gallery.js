import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Gallery extends Component {
  constructor() {
    super();
    this.state = {
      index: 0, 
      valueLeft: true,
      valueRight: false
    };
  }

  handlePreviousImage() {
    this.setState((prevState, props) => {
      const index = prevState.index - 1;
      const valueLeft = (index === 0) ? true : false;
      const valueRight = (index < props.bunnies.length) ? false : true;
      return { index, valueLeft, valueRight };
    });
  }

  handleNextImage() {
    this.setState((prevState, props) => {
      const index = prevState.index + 1;
      const valueRight = (index < (props.bunnies.length - 1)) ? false : true;
      const valueLeft = (index > 0) ? false : true;
      return { index, valueRight, valueLeft };
    });
  }


  render() {
    const { bunnies } = this.props;
    return(
      <div id="gallery-container">
        <button className="button previous" disabled={this.state.valueLeft} onClick={() => this.handlePreviousImage()}><arrow className="arrow left"></arrow>  Previous</button>
        <section>
          <img alt="bunny" src={bunnies[this.state.index].url}/>
          <figcaption>{bunnies[this.state.index].title}</figcaption>
          <figcaption>{bunnies[this.state.index].description}</figcaption>
        </section>
        <button className="button next" disabled={this.state.valueRight} onClick={() => this.handleNextImage()}>Next  <arrow className="arrow right"></arrow></button>
      </div>
    );
  }
}

Gallery.propTypes = {
  valueLeft: PropTypes.boolean,
  valueRight: PropTypes.boolean,
  handlePreviousImage: PropTypes.func,
  handleNextImage: PropTypes.func
};
