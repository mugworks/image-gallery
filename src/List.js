import React, { Component } from 'react';



export default class List extends Component {
  
  
  render() {
    const { bunnies, onRemove, onAddImg } = this.props;
    return(
      <div>
        <table id="table">
          <thead>
            <tr>
              <th>Title</th>
              <th>Description</th>
              <th></th>
            </tr>
          </thead>
          <tbody>
            {bunnies.map(bunny => <ListItem key={bunny._id} id={bunny._id} title={bunny.title} description={bunny.description} url={bunny.url} onRemove={onRemove}/>)}
          </tbody>
        </table>
        <AddImg onAddImg={onAddImg}/>
        
      </div>
    );
  }
}

class ListItem extends Component {

  render() {
    const { id, title, description, url, onRemove } = this.props;
    return(
      <tr>
        <td><a href={url}>{ title }</a></td>
        <td>{ description }</td>
        <td><button className="button remove" onClick={() => onRemove(id)}>Remove</button></td>
      </tr>
    );
  }
}


class AddImg extends Component {
  render() {
    const { onAddImg } = this.props;
    return (
      <form id="form" onSubmit={event => {
        event.preventDefault();
        const { elements } = event.target;
        const addedImage = {
          title: elements.title.value,
          description: elements.description.value,
          url: elements.url.value
        };
        onAddImg(addedImage);
        elements.title.value='';
        elements.description.value='';
        elements.url.value='';
      }}>
        <fieldset>
          <legend>Add an image</legend>
          <label>Image Title: </label>
          <input id="form-input" name="title"/>
          <label>Description: </label>
          <input id="form-input" name="description"/>
          <br/>
          <label>Url: </label>
          <input id="form-input" name="url"/>
          <button className="button add" type="submit">Add</button>
        </fieldset>
      </form>
    ); 
  }
}



