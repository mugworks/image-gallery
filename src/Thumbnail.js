import React, { Component } from 'react';

export default class Thumbnail extends Component {
  render() {
    const { bunnies } = this.props;

    return(
      <div className="thumbnail">
        {bunnies.map((bunny) => <div key={bunny._id}><figcaption>{bunny.title}</figcaption><img className="thumbnail-img"  alt={bunny.title} src={bunny.url}/></div>)}
      </div>
    );
  }
}
