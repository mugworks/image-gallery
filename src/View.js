import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { v1 } from 'uuid';
import { removeImg } from './actions';
import { addImg } from './actions';

import List from './List';
import Thumbnail from './Thumbnail';
import Gallery from './Gallery';

const Views = {
  list: List,
  thumbnail: Thumbnail,
  gallery: Gallery
};

const views = Object.keys(Views);

class View extends Component {
  
  constructor() {
    super();
    this.state = {
      bunnies: [
        {
          _id: v1(),
          title: 'Harlequin Bunny',
          description: 'This harlequin bunny artfully poses for a picture.',
          url: 'http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg',
        }, 
        {
          _id: v1(),
          title: 'Resting Rabbit',
          description: 'Ready to cuddle?',
          url: 'http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg',
        },
        {
          _id: v1(),
          title: 'Snow Bunny',
          description: 'You would have trouble finding this bunny on a snowy day.',
          url: 'http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg',
        },
      ],
      view: 'list'
    };
  }

  handleUpdateView = (view) => {
    this.setState({ view });	
  }
  
  handleRemoveImg = id => {
    const newState = removeImg(this.state, id);
    this.setState(newState);
  }

  handleAddImg = (elements) => {
    const newState = addImg(this.state, elements);
    this.setState(newState);
  }
  
  
  render() {
    const { bunnies, view } = this.state;
    const SelectedView = Views[view];
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Cute Bunnies!</h1>
        </header>
        <div>
          <p className="App-intro">
            Select which view of cute bunnies you would like.
          </p>
          {views.map((view) => <button key={view} className="button" onClick={() => this.handleUpdateView(view)}>{view[0].toUpperCase()+view.slice(1)}</button>)}
          
          
          <SelectedView bunnies={bunnies} onRemove={this.handleRemoveImg} onAddImg={this.handleAddImg}/>
        </div>
      </div>
    );
  }
}

export default View;

View.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  url: PropTypes.string,
  handleAddImg: PropTypes.func,
  handleRemoveImg: PropTypes.func,
  handleUpdateView: PropTypes.func,
  view: PropTypes.string
};