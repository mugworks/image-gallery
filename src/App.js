import React, { Component } from 'react';

import './App.css';
import View from './View';

	
class App extends Component {	
  
	
	
	
	render(){
	  return(
	    <View/>
	  );
	}
}
export default App;
