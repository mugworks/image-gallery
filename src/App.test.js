import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';
import View from './View';
import Gallery from './Gallery';
import Thumbnail from './Thumbnail';
import List from './List';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('renders a snapshot', () => {
  // View.handleUpdateView('gallery');
  const tree = renderer.create(<View/>).toJSON();
  expect(tree).toMatchSnapshot();
});

// it ('renders a snapshot', () => {
//   const tree = renderer.create(<List/>).toJSON();
//   expect(tree).toMatchSnapshot();
// });

// import React from 'react';
// import { shallow } from 'enzyme';
// import toJSON from 'enzyme-to-json';
// import View from './View';

// describe('App', () => {

//   it('should render View View', () => {
//     const wrapper=shallow(<View/>);
//     expect(toJSON(wrapper)).toMatchSnapshot();
//   });
// });

// import React from 'react';
// import { shallow, mount } from 'enzyme';
// import toJSON from 'enzyme-to-json';
// import { Gallery } from './Gallery';
// import View from './View';


// describe('App', () => {

  // it('should render Gallery View', () => {
  //   const wrapper=shallow(<Gallery valueLeft="true"/>);
  //   expect(toJSON(wrapper)).toMatchSnapshot();
  // });

//   it('renders a snapshot', () => {
//     // View.handleUpdateView('gallery');
//     const wrapper = shallow(<View/>);
//     expect(toJSON(wrapper)).toMatchSnapshot();
//   });
// });

