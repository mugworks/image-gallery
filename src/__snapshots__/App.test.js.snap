// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Cute Bunnies!
    </h1>
  </header>
  <div>
    <p
      className="App-intro"
    >
      Select which view of cute bunnies you would like.
    </p>
    <button
      className="button"
      onClick={[Function]}
    >
      List
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      Thumbnail
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      Gallery
    </button>
    <div>
      <table
        id="table"
      >
        <thead>
          <tr>
            <th>
              Title
            </th>
            <th>
              Description
            </th>
            <th />
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <a
                href="http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg"
              >
                Harlequin Bunny
              </a>
            </td>
            <td>
              This harlequin bunny artfully poses for a picture.
            </td>
            <td>
              <button
                className="button remove"
                onClick={[Function]}
              >
                Remove
              </button>
            </td>
          </tr>
          <tr>
            <td>
              <a
                href="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg"
              >
                Resting Rabbit
              </a>
            </td>
            <td>
              Ready to cuddle?
            </td>
            <td>
              <button
                className="button remove"
                onClick={[Function]}
              >
                Remove
              </button>
            </td>
          </tr>
          <tr>
            <td>
              <a
                href="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg"
              >
                Snow Bunny
              </a>
            </td>
            <td>
              You would have trouble finding this bunny on a snowy day.
            </td>
            <td>
              <button
                className="button remove"
                onClick={[Function]}
              >
                Remove
              </button>
            </td>
          </tr>
        </tbody>
      </table>
      <form
        id="form"
        onSubmit={[Function]}
      >
        <fieldset>
          <legend>
            Add an image
          </legend>
          <label>
            Image Title: 
          </label>
          <input
            id="form-input"
            name="title"
          />
          <label>
            Description: 
          </label>
          <input
            id="form-input"
            name="description"
          />
          <br />
          <label>
            Url: 
          </label>
          <input
            id="form-input"
            name="url"
          />
          <button
            className="button add"
            type="submit"
          >
            Add
          </button>
        </fieldset>
      </form>
    </div>
  </div>
</div>
`;
